blockchain/chain_sync/src/sync.rs

```diff
@@ -168,27 +170,40 @@ where
                     }
                 }
                 NetworkEvent::PeerDialed { peer_id } => {
-                    let heaviest = self.chain_store.heaviest_tipset().unwrap();
-                    self.network
+                    let heaviest = chain_syncer.chain_store.heaviest_tipset().await.unwrap();
+                    chain_syncer.network
                         .hello_request(
                             peer_id,
                             HelloMessage {
                                 heaviest_tip_set: heaviest.cids().to_vec(),
                                 heaviest_tipset_height: heaviest.epoch(),
                                 heaviest_tipset_weight: heaviest.weight().clone(),
-                                genesis_hash: self.genesis.blocks()[0].cid().clone(),
+                                genesis_hash: chain_syncer.genesis.blocks()[0].cid().clone(),
                             },
                         )
                         .await
                 }
+                NetworkEvent::PubsubMessage { source, topics, message } => {
+                    match topics[0].as_str() {
+                       "/fil/blocks/interop"  => {
+                           let new_blk: BlockHeader = BlockHeader::unmarshal_cbor(&message).unwrap();
+                           let new_blk = Tipset::new(vec![new_blk]).unwrap();
+                           let chain_syncer = chain_syncer.clone();
+                           task::spawn(async move {
+                               chain_syncer.sync(&new_blk).await.unwrap();
+                           });
```

Comment:
I really don't think this is what we want to do, we shouldn't be manually constructing the tipset and doing a full sync from it, can you explain reasoning?

Reply:
Hmm, I think we actually are supposed to call inform_new_head here instead. Does that answer your question?

Reply:
yeah, that makes a lot more sense, calling sync I'm pretty sure isn't what we want